packages/@n8n/decorators/src/module.ts

```diff
@@ -1,7 +1,7 @@
 import { Container, Service, type Constructable } from '@n8n/di';
 import type { ExecutionLifecycleHooks } from 'n8n-core';
 
-import type { MultiMainSetup } from '@/scaling/multi-main-setup.ee';
+import type { MultiMainSetup } from './types';
```

Comment:
We should decouple the metadata state from the actual registration here as well.
Then we don't need `MultiMainSetup` in this package.

Reply:
If it's okay, I've done [this](https://github.com/n8n-io/n8n/pull/14784/commits/7b452d7dc424bf076560ae2307515d097f2d452a#diff-a2617634b9c628bb71defd2d61d715ab30342136572746eff4c11bc44d54cda2) temporarily, so I can clear the way to `@n8n/db` to unblock extensions. Once that's done we can come back to add more decorators?